features/puppet/defaults.pan

```diff
@@ -11,5 +11,16 @@ variable PUPPET_FULLVERSION = {
     version;
 };
 
-include if (pkg_compare_version(PUPPET_FULLVERSION, '4.0.0') != PKG_VERSION_LESS) 'features/puppet/defaults_4+';
-include if (pkg_compare_version(PUPPET_FULLVERSION, '5.0.0') != PKG_VERSION_LESS) 'features/puppet/defaults_5+';
+variable PUPPET_INCLUDE = {
+    if (pkg_compare_version(PUPPET_FULLVERSION, '6.0.0') <= 0) {
+        'features/puppet/defaults_6+';
+    } else if (pkg_compare_version(PUPPET_FULLVERSION, '5.0.0') <= 0) {
+        'features/puppet/defaults_5+';
+    } else if (pkg_compare_version(PUPPET_FULLVERSION, '4.0.0') <= 0) {
+        'features/puppet/defaults_4+';
+    } else {
+        null;
+    }
+};
+
```

Comment:
This change does not just add 6+ but actually changes the way 5+ and 4+ are used. In the original code defaults_4+ is included for all releases equal or above 4, and similarly for default_5+. Here defaults_4+ is included for the 4.* releases, 5+ for the 5.* releases, etc. The effect is that if you are running puppet 5, the defaults_4+ is not included and and this makes some defaults values  wrong. I'm fine with changing the logic but this requires changing defaults_4+ and defaults_5+.  Or keep the same logic and add one line similar to the original ones.

Reply:
@jrha sorry the review of these changes somehow slipped out of my todo list. But I had everything ready for testing and I'm doing it now.

Reply:
No problem, thanks for taking a look at it!

Reply:
@sartiran did you find time to test, or shall I push this back until next year?

Reply:
I've tested on Oct 9 and put some comments (above and below). These changes are not backward compatibles. But I've seen no answers.

Reply:
Okay thanks!